sdk/python/kfp/compiler/test_data/pipelines/pipeline_with_custom_artifact_type.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2022 The Kubeflow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import sys
+import tempfile
+
+
+def create_temporary_non_kfp_artifact_package(
+        temp_dir: tempfile.TemporaryDirectory) -> None:
+    import inspect
+    import os
+    import textwrap
+
+    class ThirdPartyModel:
+        schema_title = 'third_party.ThirdPartyModel'
```

Comment:
This won't work. Schema title must reference some known artifact type that's pre-registered with MLMD backend. 
We need to stick with existing google namespaced types.

Reply:
Agreed. This was included to test the compilation logic, not execution logic. There are no non-KFP libraries that have custom artifact classes that are compliant with KFP v2 artifacts.

Do you think adding a comment to indicate that this is a compilation test, not an execution test/example is sufficient until a non-KFP library adds compliant types?

Reply:
The compilation result shows `system.Artifact` which is incorrect if `third_party.ThirdPartyModel` were registered. Why not just use some existing google type, which make this sample executable today? Are you concerned about tinting the sample with some platform-specific elements?

Reply:
I've updated the compilation logic -- waiting for tests to pass.

Is there currently a non-kfp library that implements a custom artifact type that is compatible with KFP artifacts and can be installed in the same environment as kfp v2? I do know of one, but would readily include one if it exists. I am not concerned about tinting the sample.